### ISSUE TITLE
Support for NVIM_APPNAME

### ISSUE THREAD

**Issue by oldbuilding** [AstroNvim/AstroNvim#1677]
### Checklist

- [X] Using a stable version of Neovim (i.e. not neovim nightly)
- [X] `:AstroUpdate`
- [X] Restarted AstroNvim

### Operating system/version

Windows 10

### Terminal/GUI

Alacritty

### AstroNvim Health

AstroNvim
  3   - AstroNvim Version: v3.4.2
  4   - Neovim Version: v0.9.0-dev-3922+gd8f9dcd5d-dirty
  5   - WARNING Neovim nightly is not officially supported and may have breaking changes
  6   - OK git is installed: Used for core functionality such as updater and plugin management
  7   - OK xdg-open is installed: Used for gx mapping for opening files with system opener (Optional)
  8   - WARNING lazygit is not installed: Used for mappings to pull up git TUI (Optional)
  9   - OK node is installed: Used for mappings to pull up node REPL (Optional)
 10   - WARNING gdu is not installed: Used for mappings to pull up disk usage analyzer (Optional)
 11   - WARNING btm is not installed: Used for mappings to pull up system monitor (Optional)
 12   - OK python3 is installed: Used for mappings to pull up python REPL (Optional)

### Describe the bug

AstroNvim assumes that the config path should end with "astronvim".  This leads to the following scenario when the config path already ends with "astronvim"

```
astro install {  home = "/home/<user>/.config/astronvim" }

I believe the issue is in `bootstrap.lua` on line 18.

```
 astronvim.install = _G["astronvim_installation"] or { home = vim.fn.stdpath "config" }                                                                                                                             
 print("astro install " .. vim.inspect(astronvim.install))
```
> astro install {  home = "/home/<user>/.config/astronvim" }

```
  astronvim.supported_configs = { astronvim.install.home }
  print("astro install home " .. vim.inspect(astronvim.install.home))  
```

This is correct.
> "/home/<user>/.config/astronvim"

**line 18**
``` 
 astronvim.install.config = vim.fn.stdpath("config"):gsub("nvim$", "astronvim")
  print("astro install config " .. vim.inspect(astronvim.install.config)) 
```
Mangled string replacement.
> "/home/<user>/.config/astroastronvim"



### Steps to Reproduce

run nvim with the following

```
NVIM_APPNAME=astronvim nvim
```

### Expected behavior

astro install config path should respect NVIM_APPNAME, which will default to "nvim" if not set.

I experimented with a few ideas and was able to get the path set correctly just by removing `:gsub("^nvim$", "astronvim").

I also tried setting the environment variable inline but then realized that approach doesn't make sense.  If users are starting their astronvim instance of nvim by simply calling `nvim`, the environment is already correct.  They may have cloned astronvim into a folder called ~/.config/nvim, or done like me and cloned into a separate directory called ~/.config/astronvim.


### Screenshots

_No response_

### Additional Context

_No response_

**Comment by mehalter** [comment]
Yeah this is definitely a bug. On the bright side, this will still work for your needs totally fine :) it will just also accept configurations living in `. config/astroastronvim` but in the mean time just putting the configuration in `~/.config/astronvim` and using NVAPP will work.

**Comment by mehalter** [comment]
But yeah I will definitely fix the secondary configuration location

**Comment by oldbuilding** [comment]
Thank you!  I just updated and confirmed.  Looks good now!
